### PR TITLE
xWidthAvg: Update character frequency weightings data source

### PR DESCRIPTION
**Pull request by michaeltaranto**
Character frequency weightings used to calculate the `xWidthAvg` metrics were previously hard coded internally, and were an adaption from a [frequency table] on Wikipedia.

We now generate these weightings based on the abstracts of [WikiNews] articles, making it possible to add support for other languages that make use of non-latin [unicode subsets], e.g. Thai.

The updated `xWidthAvg` metrics are very close to the original hard coded values.
This results in either no or extremely minor changes to the generated fallback font CSS, meaning we don't expect any notable changes to consumers, with the benefit being this lays the ground work to support additional language subsets in the future.

[frequency table]: https://en.wikipedia.org/wiki/Letter_frequency#Relative_frequencies_of_letters_in_other_languages
[WikiNews]: https://wikinews.org/
[unicode subsets]: https://www.utf8icons.com/subsets

---

### Note for the reviewer
This PR is in preparation for a follow up PR that introduces unicode subset support. So some refactoring in this PR is in service of a cleaner diff for the next PR.

### CODE REVIEW

File: packages/metrics/scripts/build.ts [hunk 1]
```diff
@@ -4,18 +4,16 @@ import dedent from 'dedent';
...
 import { fontFamilyToCamelCase } from './../src';
+import { buildMetrics } from './buildMetrics';
 
 const writeFile = async (fileName: string, content: string) =>
   await fs.writeFile(path.join(__dirname, fileName), content, 'utf-8');
 
-interface MetricsFont extends Font {
-  category: string;
-}
+type MetricsFont = Awaited<ReturnType<typeof buildMetrics>>;
```
Comment (askoufis) [inline packages/metrics/scripts/build.ts:16]: This type is identical to the one in `buildMetrics.ts`. Any reason not to just export it?

File: packages/unpack/scripts/build-weightings.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      let filteredTotal = 0;
+
+      const output = Object.keys(charOccurenceCount)
+        .map((char) => ({
+          char,
+          count: charOccurenceCount[char],
+          weighting: parseFloat(
+            (charOccurenceCount[char] / rawTotal).toFixed(4),
+          ),
+        }))
```
Comment (askoufis) [inline packages/unpack/scripts/build-weightings.ts:65]:
```suggestion
      const output = Object.entries(charOccurenceCount)
        .map(([char, count]) => ({
          char,
          count,
          weighting: parseFloat((count / rawTotal).toFixed(4)),
        }))
```

File: packages/metrics/scripts/build.ts (hunk 1 above)
Comment (michaeltaranto) [reply packages/metrics/scripts/build.ts:16]: Yeah fair play 👌

File: package.json [hunk 3]
```diff
@@ -10,8 +10,9 @@
     "test": "jest",
     "format": "prettier --write .",
     "lint": "manypkg check && prettier --check . && tsc",
-    "dev": "pnpm %packages dev",
-    "build": "pnpm %packages build && pnpm metrics:generate",
+    "dev": "pnpm %packages dev && pnpm generate",
+    "build": "pnpm generate && pnpm %packages build && pnpm metrics:generate",
+    "generate": "pnpm unpack:generate && pnpm metrics:generate",
```
Comment (askoufis) [inline package.json:15]: This looks like it's going to run `metrics:generate` twice: once as part of `pnpm generate` and then again at the end.

**Review by askoufis — APPROVED** [review]

**Pull request merged**